dev/format_check.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+"""Erlang formatter for CouchDB
+Warning: this file needs to run from the CouchDB repo root.
+USAGE: ERLFMT_PATH=<path_to_erlfmt> python3 dev/format_check.py
+"""
+
+import os
+import subprocess
+import sys
+
+from format_lib import get_source_paths, get_erlang_version
+
+FILTERED_LINES = [
+    "Checking formatting...",
+    "[warn] Code style issues found in the above file(s). Forgot to run erlfmt?",
+    "",
+]
+
+
+if __name__ == "__main__":
+    if get_erlang_version() < 21:
+        print("Erlang version is < 21. Skipping format check")
+        sys.exit(0)
+
+    failed_checks = 0
+    for item in get_source_paths():
+        run_result = subprocess.run(
+            [
+                os.environ["ERLFMT_PATH"],
+                "-c",
+                "--verbose",
+                # We have some long lines and erlfmt doesn't forcefully wrap
+                # them all. We should decrease this over time
+                "--print-width=167",
+                item["raw_path"],
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        if run_result.returncode != 0:
+            # erlfmt sometimes returns a non-zero status code with no
+            # actual errors. This is a workaround
```

Comment:
These are still happening but I think it's partly from our slight misunderstanding of erlfmt logic.

We format lines with a plain command `erlfmt -w $path` but when we check we use `--print-width=167` because we want to avoid the warning about some string being too long to wrap. However this also means that we now get a non-0 exit code.

For example formatting `src/chttpd/src/chttpd_auth_cache.erl`
```
% ./bin/erlfmt --verbose -w src/chttpd/src/chttpd_auth_cache.erl
Formatting src/chttpd/src/chttpd_auth_cache.erl
src/chttpd/src/chttpd_auth_cache.erl:128: Line too long (135 > 100)
```

Checking with default args:
```
% ./bin/erlfmt --verbose -c src/chttpd/src/chttpd_auth_cache.erl
Checking formatting...
Formatting src/chttpd/src/chttpd_auth_cache.erl
src/chttpd/src/chttpd_auth_cache.erl:128: Line too long (135 > 100)
All matched files use erlfmt code style!

% echo $?
0
```

Checking the way we currently check:
```
% ./bin/erlfmt --verbose -c src/chttpd/src/chttpd_auth_cache.erl --print-width=167
Checking formatting...
Formatting src/chttpd/src/chttpd_auth_cache.erl
[warn] src/chttpd/src/chttpd_auth_cache.erl
[warn] Code style issues found in the above file(s). Forgot to run erlfmt?

% echo $?
1
```

So I think that solves the mystery. I'll try to remove the check.